File: js/15649.js
const fs = require('fs');

const [n, m] = fs
  .readFileSync('/dev/stdin')
  .toString()
  .trim()
  .split(' ')
  .map(Number);

const isVisited = Array(n + 1).fill(false);
const arr = [];
let result = '';

dfs(0);
console.log(result);

function dfs(cnt) {
  if (cnt === m) {
    result += `${arr.join(' ')}\n`;
    return;
  }
  for (let i = 1; i <= n; i++) {
    if (!isVisited[i]) {
      isVisited[i] = true;
      arr.push(i);
      dfs(cnt + 1);
      arr.pop();
      isVisited[i] = false;
    }
  }
}

File: c++/15649.cpp
#include <iostream>

using namespace std;

int n, m;
int arr[9];
bool isVisited[9] = {false,};

void dfs(int cnt){
  if(cnt==m){
    for(int i=0; i<m; i++){
      cout << arr[i] << ' ';
    }
    cout << '\n';
    return;
  }
  for(int i=1; i<=n; i++){
    if(!isVisited[i]){
      isVisited[i] = true;
      arr[cnt]=i;
      dfs(cnt+1);
      isVisited[i] = false;
    }
  }
}

int main(){
  cin >> n >> m;
  dfs(0);
  return 0;
}